### ISSUE TITLE
QUESTION: Should the design for aic/bic reported by OLS be changed

### ISSUE THREAD

**Issue by wrapper228** [statsmodels/statsmodels#8284]
AIC is a relative measure used for model selection - and for feature selection in particular. It's confusing that AIC can be reported by a single model - it is useless and misleading for feature selection task.
Right approach for AIC feature selection algorithm implies to compute variance on all features included and then (for each subset of features) to compute model log likelihood using that variance (computed on all features included). If user does not know that, then he can solve feature selection task by computing AIC on individual models, which is absolutely wrong for feature selection (but still looks intuitively correct from API point of view).

#### Example of code that reproduces wrong solution for feature selection task:

```python
import pandas as pd
import statsmodels.api as sm
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from itertools import combinations

X, y = make_classification(n_samples=2000, n_features=10, n_informative=5, n_redundant=5,
                           flip_y=0.05, random_state=0)

X = pd.DataFrame(X, columns=list(range(10)))
y = pd.Series(y)

train_X, test_X, train_y, test_y = train_test_split(X, y, train_size=1500, random_state=1)

best_feature_set = []
best_aic = 0

for i in range(train_X.shape[1]):
    for subset in combinations(train_X.columns.tolist(), i+1):

        model = sm.OLS(train_y, train_X[list(subset)])
        
        this_aic = model.fit().aic
        
        if this_aic > best_aic:
            best_feature_set = subset
            best_aic = this_aic
            
print("best feature set is:", best_feature_set, " with AIC ", best_aic)
```

**Comment by josef-pkt** [comment]
Our aic and bic are the standard definition for loglikelihood based information criteria.

These information criteria **can** be compared across models as long as they have the same endogenous target variable.

I'm not sure what you are saying, but there are other selection criteria besides the standard information criteria.  But if we don't have those, then it's a missing enhancement and not a problem with the definition of aic, bic, ...

(Another definition for aic and bic in the linear regression case is based on residual sum of squares available in `eval_measures` module. However, those definitions don't include the full loglikelihood and therefore are not comparable across models anymore, just for feature selection within a model class.)

For example, penalization parameter selection using aic with effective degrees of freedom is our main choice for selecting the penalization parameter in our GAM.

**Comment by bashtage** [comment]
> AIC is a relative measure used for model selection - and for feature selection in particular. It's confusing that AIC can be reported by a single model - it is useless and misleading for feature selection task.

Yes, it is not particularly meaningful for a single model.  IT is attached to results to let users capture its value so that they can use it further down the line.  It is printed as part of a summary so that users who which do perform interactive modeling can rapidly compare AIC values simply by checking the console and doing some mental math. 

There is some expectation that users have a reasonable understanding of statistics. While ti tries to make reasonable default choices (e.g., homoskedastic errors for regression), it needs to present a wide range of features to let users write custom code to accomplish their goals.


There are a couple of mistakes in your code.  

1. AIC should be as small as possible, so the check should be `this_aic < best_aic`. 
2. AIC can take any value, so `best_aic = np.inf` should be used to initialize the algo.
